pkgs/servers/jetbrains/youtrack/default.nix

```diff
@@ -0,0 +1,85 @@
+{ lib, stdenv, fetchurl, fetchzip, makeWrapper, jdk17, p7zip, gawk, baseStatePath ? "/var/lib/youtrack" }:
+let
+  meta = with lib; {
+    description = "Issue tracking and project management tool for developers";
+    maintainers = teams.serokell.members ++ [ lib.maintainers.leona ];
+    sourceProvenance = with sourceTypes; [ binaryBytecode ];
+    # https://www.jetbrains.com/youtrack/buy/license.html
+    license = licenses.unfree;
+  };
+
+  generic = {
+    version, hash,
+  }: stdenv.mkDerivation rec {
+    pname = "youtrack";
+    inherit version;
+
+    # This base version is seperated by _ instaed of . to ensure path compatability and confusion
+    baseVersion = with lib; pipe version [
+      versions.splitVersion
+      (take 2)
+      (concatStringsSep "_")
+    ];
+
+    src = fetchzip {
+      url = "https://download.jetbrains.com/charisma/youtrack-${version}.zip";
+      inherit hash;
+    };
+
+    nativeBuildInputs = [ makeWrapper p7zip ];
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/app
+      cp -a . $out/app
+
+      for path in "backups" "conf" "data" "logs" "temp"
+      do
+        rm -rf $out/app/$path
+        ln -s ${baseStatePath}/${baseVersion}/$path $out/app/$path
+      done
+
+      makeWrapper "$out/app/bin/youtrack.sh" "$out/bin/youtrack" \
+        --prefix PATH : "$out/libexec/app:${lib.makeBinPath [ jdk17 gawk ]}" \
+        --set FJ_JAVA_EXEC "${jdk17}/bin/java" \
```

Comment:
Is there anything this fixes? Because I don't set this var and it works perfectly well